### ISSUE TITLE
Incorrect wrapping into `apply` function

### ISSUE THREAD

**Issue by petertrr** [analysis-dev/diktat#1043]
## Describe the bug
```kotlin
fun fo() {
    val pair = it.split("=", limit = 2).map {
        it.replace("\\=", "=")
    }
    pair.first() to pair.last()
}
```

## Expected behavior:
Rule shouldn't trigger, because top-level node is an infix function, not dot-qualified call, from which we can extract receiver.
Also, it can potentially break in other cases, when result of dot-qualified call is implicitly returned. If we wrap it into `apply`, we will break the code, while wrapping into `run` will not.

## Observed behavior
```kotlin
fun fo() {
    val pair = it.split("=", limit = 2).map {
        it.replace("\\=", "=")
    }.apply {
        first() to pair.last()
    }
}
```

## Environment information
* diktat version: 1.0.0-rc.3

**Comment by kgevorkyan** [comment]
Maybe, after https://github.com/cqfn/diKTat/pull/1036, the observed behavior will be

```kotlin
fun fo() {
    val pair = it.split("=", limit = 2).map {
        it.replace("\\=", "=")
    }.apply {
        first() to last()
    }
}
``` 

but it's should be checked

**Comment by akuleshov7** [comment]
This affects check. Need to cover this also in 1.0.4
